packages/typespec-client-generator-core/README.md

```diff
@@ -291,6 +292,40 @@ op nameInService: void;
 op nameInService: void;
 ```
 
+#### `@clientNamespace`
+
+Changes the namespace of a client, or model generated in the client SDK
+
+```typespec
+@Azure.ClientGenerator.Core.clientNamespace(rename: valueof string, scope?: valueof string)
+```
+
+##### Target
+
+`unknown`
+
+##### Parameters
+
+| Name   | Type             | Description                                                                                                   |
+| ------ | ---------------- | ------------------------------------------------------------------------------------------------------------- |
+| rename | `valueof string` | The rename you want applied to the object                                                                     |
+| scope  | `valueof string` | The language scope you want this decorator to apply to. If not specified, will apply to all language emitters |
+
+##### Examples
+
+```typespec
+@clientNamespace("ContosoClient")
+namespace Contoso;
+```
+
+```typespec
+@clientName("ContosoJava", "java")
+@clientName("ContosoPython", "python")
+@clientName("ContosoCSharp", "csharp")
+@clientName("ContosoJavascript", "javascript")
+namespace Contoso;
```

Comment:
the `@clientNamespace` decorator is needed because we can't differentiate very well between the client's name and the namespace's name, is this correct?

If we wanted to keep the namespace as contoso while changing each language's client's names, would this work as well?

```tsp
@client({name: "ContosoJava"}, "java")
@client({name: "ContosoPython"}, "python")
@client({name: "ContosoCsharp"}, "csharp")
@client({name: "ContosoJavascript"}, "javascript")
namespace Contoso;
```

I agree though that even if the above code works, it's unfortunate that we have `@client{name}` and `@clientName`, and I am open to adding `@clientNamespace` for clarity

Reply:
if we rename the client name with `@client`, then we need to redeclare all the client hierarchy.
another reason the add `@clientNamespace` is we could use it to move a model/enum/union to another namespace separately.